### PR TITLE
GCP wrapper

### PR DESCRIPTION
**Pull request by aroundthecode**
Add a GCP wrapper to work on Google SQL instances

In reference to #82 and #63 I tried to provide an initial GCP implementation.

To have operator work with GCP you have 
1) use postgres connection in secret
2) manually create a Master role e.g. "devops-operators" (must NOT be the same as the connection secret)
3) use such role in database CR e.g. spec.masterRole: devops-operator

DropRole method will check for db owner and will skip master role dropping.

I'm not a GO developer so I tried to keep all modification in gcp.go, please modify at will to make it compliant with other code style 

Working with both Database and Role CR, both in creation and deletion.
@hitman99

### CODE REVIEW

**Comment by hitman99** [comment]
Thank you for your contribution, I will get this reviewed today

File: pkg/postgres/gcp.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
+package postgres
+
+/*
+To have operator work with GCP you have
```
Comment (hitman99) [inline pkg/postgres/gcp.go:4]: Would you mind adding these to the README.md so people would not have to dig in code comments ?

File: pkg/postgres/gcp.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		return err
+	}
+	_, err = c.db.Exec(fmt.Sprintf(DROP_DATABASE, database))
+	// Error code 3D000 is returned if database doesn't exist
+	if err != nil && err.(*pq.Error).Code != "3D000" {
+		return err
+	}
+
+	logger.Info(fmt.Sprintf("Dropped database %s", database))
+
```
Comment (hitman99) [inline pkg/postgres/gcp.go:49]: remove empty line

File: pkg/postgres/gcp.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+		return err
+	}
+	return nil
+}
+
+
+func (c *gcppg) DropRole(role, newOwner, database string, logger logr.Logger) error {
+	
+	tmpDb, err := GetConnection(c.user, c.pass, c.host, database, c.args, logger)
+	q := fmt.Sprintf("SELECT pg_catalog.pg_get_userbyid(d.datdba) FROM pg_catalog.pg_database d WHERE d.datname = '%s';", database)
```
Comment (hitman99) [inline pkg/postgres/gcp.go:71]: This should be in postgres.go and not hardcoded here

File: pkg/postgres/gcp.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	if err != nil {
+		return err
+	}
+	var masterRole string
+	for rows.Next() {
+		rows.Scan(&masterRole)
+	}
+
+	if( role != masterRole){
+
```
Comment (hitman99) [inline pkg/postgres/gcp.go:83]: remove empty line

File: pkg/postgres/gcp.go [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+	if( role != masterRole){
+		
+		q = fmt.Sprintf(DROP_ROLE, role)
+		logger.Info("GCP Drop Role: "+ q)
+		_, err = tmpDb.Exec(q)
+		// Check if error exists and if different from "ROLE NOT FOUND" => 42704
+		if err != nil && err.(*pq.Error).Code != "42704" {
+			return err
+		}
+
```
Comment (hitman99) [inline pkg/postgres/gcp.go:91]: remove empty line

**Review by hitman99 — CHANGES_REQUESTED** [review]

**Comment by aroundthecode** [comment]
@hitman99 any chance you can review this shortly and generate a new version?
I'm using a patched operator version, but I'd prefer to stay on the mainstream :)

**Review by hitman99 — APPROVED** [review]

**Comment by hitman99** [comment]
@aroundthecode PR approved and merged. Thanks for using my operator :)

**Pull request merged**